sdk/monitor/monitor-ingestion/src/logsIngestionClient.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { TokenCredential } from "@azure/core-auth";
+import { CommonClientOptions } from "@azure/core-client";
+import { SDK_VERSION } from "./constants";
+import { GeneratedDataCollectionClient } from "./generated";
+import { FailedLogsIngestionError, UploadOptions, UploadResult } from "./models";
+import { GZippingPolicy } from "./gZippingPolicy";
+import { asyncPool } from "./concurrentPoolHelper";
+
+/**
+ * Options for Montior Logs Ingestion Client
+ */
+export interface LogsIngestionClientOptions extends CommonClientOptions {
+  /** Api Version */
+  apiVersion?: string;
+}
+const defaultIngestionScope = "https://monitor.azure.com//.default";
+
+/**
+ * Client for Monitor Logs Ingestion
+ */
+export class LogsIngestionClient {
+  /**
+   * Overrides client endpoint.
+   */
+  private endpoint: string;
+  private _dataClient: GeneratedDataCollectionClient;
+  /**
+   * Construct a MonitorIngestionClient that can be used to query logs using the Log Analytics Query language.
+   *
+   * @param tokenCredential - A token credential.
+   * @param options - Options for the MonitorIngestionClient.
+   */
+  constructor(
+    endpoint: string,
+    tokenCredential: TokenCredential,
+    options?: LogsIngestionClientOptions
+  ) {
+    const credentialOptions = {
+      credentialScopes: defaultIngestionScope,
+    };
+    this.endpoint = endpoint;
+    const packageDetails = `azsdk-js-monitor-ingestion/${SDK_VERSION}`;
```

Comment:
userAgentPrefix is already handled in the generated client context class so should be removed here.

Reply:
The generated client has this - `azsdk-js-data-collection/1.0.0-beta.1` instead of monitor-ingestion. Is that ok?

Reply:
there should be a way to customize the package name in swagger readme file.

Reply:
done